pkg/providers/subnet/subnet.go

```diff
@@ -193,19 +181,19 @@ func (p *DefaultProvider) UpdateInflightIPs(createFleetInput *ec2.CreateFleetInp
 		if !lo.Contains(subnetIDsToAddBackIPs, *cachedSubnet.SubnetId) {
 			continue
 		}
-		originalSubnet, ok := lo.Find(subnets, func(subnet *ec2.Subnet) bool {
-			return *subnet.SubnetId == *cachedSubnet.SubnetId
+		originalSubnet, ok := lo.Find(subnets, func(subnet v1beta1.Subnet) bool {
+			return subnet.ID == *cachedSubnet.SubnetId
 		})
 		if !ok {
 			continue
 		}
 		// If the cached subnet IP address count hasn't changed from the original subnet used to
 		// launch the instance, then we need to update the tracked IPs
-		if *originalSubnet.AvailableIpAddressCount == *cachedSubnet.AvailableIpAddressCount {
+		if p.inflightIPs[originalSubnet.ID] == *cachedSubnet.AvailableIpAddressCount {
```

Comment:
I think this is wrong (though I need to understand it better). The old code checked if the IP addresses in the cache are the same as the subnets that we stored. This code basically does the opposite of that. I know that this isn't correct, but I'm not sure what the correct behavior is here. We should discuss more and try to work it out.

Reply:
Discussed offline, and we will be keeping the subnet logic consistent.